### PR TITLE
fix: handle backend errors in pool page

### PR DESCRIPTION
**Pull request by agualis**
Before this fix any backend error in a pool query would crash the app and send an error to sentry.

Screenshot 2024-08-29 at 17 07 10

After this fix: 
- We detect invalid pool id error. 
- We capture other errors in sentry
- We show an alert instead of crashing the app:

error

### CODE REVIEW

File: app/(app)/pools/[chain]/[variant]/[id]/layout.tsx [hunk 1]
```diff
@@ -54,12 +61,31 @@ export async function generateMetadata({
 export default async function PoolLayout({ params: { id, chain, variant }, children }: Props) {
   const _chain = slugToChainMap[chain]
 
-  const { data } = await getPoolQuery(chain, id)
+  const { data, error } = await getPoolQuery(chain, id)
 
-  if (!data.pool) {
-    return <Box>Pool with id not found ({id})</Box>
+  if (error) {
+    if (error?.message === 'Pool with id does not exist') {
```
[the remark this answers is not in the document]
Comment (agualis) [reply app/(app)/pools/[chain]/[variant]/[id]/layout.tsx:67]: As far as I know we are not receiving any error code here.

[the remark this answers is not in the document]
Comment (garethfuller) [reply app/(app)/pools/[chain]/[variant]/[id]/layout.tsx:67]: We should get a 404 from the request response. We should use that as Anastas points out. This doesn't provide an example, but if we could try and follow Next.js patterns for error handling, that would be great: https://nextjs.org/docs/app/building-your-application/routing/error-handling

[the remark this answers is not in the document]
Comment (garethfuller) [reply app/(app)/pools/[chain]/[variant]/[id]/layout.tsx:67]: Yeah agree we need to get something that at least handles the error in the meantime.

File: app/(app)/pools/[chain]/[variant]/[id]/layout.tsx [hunk 2]
```diff
@@ -54,12 +61,31 @@ export async function generateMetadata({
...
+      const error = `Pool with id not found in ${chain} (${id})`
```
Comment (garethfuller) [inline app/(app)/pools/[chain]/[variant]/[id]/layout.tsx:68]:
```suggestion
      const error = `Pool not found on ${chain}, ID (${id})`
```

File: app/(app)/pools/[chain]/[variant]/[id]/layout.tsx [hunk 3]
```diff
@@ -54,12 +61,31 @@ export async function generateMetadata({
...
+          <BalAlert status="error" content={error} ssr />
+        </DefaultPageContainer>
+      )
+    }
+    captureException(error, { level: 'fatal' })
+    return (
+      <DefaultPageContainer>
+        <BalAlert
+          status="warning"
+          content="Our API data provider appears to be having issues..."
```
Comment (garethfuller) [inline app/(app)/pools/[chain]/[variant]/[id]/layout.tsx:80]:
```suggestion
          content="Failed to fetch pool"
```

**Review by garethfuller — CHANGES_REQUESTED** [review]

**Review by garethfuller — COMMENTED** [review]
@agualis I'm struggling to get this to work how I wanted with Error boundaries. It works in dev, but in production it blocks you from rendering error messages that are thrown in server components for security reasons, which makes sense.

I need to move on to something else for now, maybe you could take another look. Maybe you were right to start with by just rendering an error message on the RSC.

**Pull request merged**